File: 0969-number-of-recent-calls/0969-number-of-recent-calls.js

var RecentCounter = function() {
    this.arr = [null];
};

/** 
 * @param {number} t
 * @return {number}
 */
RecentCounter.prototype.ping = function(t) {
    this.arr.push(t);
    let range = [t - 3000, t],
        counter = 0;
    for (let i = 1; i < this.arr.length; i++) {
        if (this.arr[i] >= range[0] && this.arr[i] <= range[1]) {
            counter++;
        }
    }
    return counter;
};

/** 
 * Your RecentCounter object will be instantiated and called as such:
 * var obj = new RecentCounter()
 * var param_1 = obj.ping(t)
 */

File: 0969-number-of-recent-calls/0969-number-of-recent-calls.py
class RecentCounter:

    def __init__(self):
        self.arr = []

    def ping(self, t: int) -> int:
        self.arr.append(t)

        while t - self.arr[0] > 3000:
            self.arr.pop(0)
        
        return len(self.arr)
            
        
                


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)